worlds/undertale/__init__.py

```diff
@@ -99,110 +116,107 @@ def create_items(self):
             itempool += [name] * num
         for name, num in non_key_items.items():
             itempool += [name] * num
-        if self.multiworld.rando_item_button[self.player]:
+        if self.options.rando_item_button:
             itempool += ["ITEM"]
         else:
             self.multiworld.push_precollected(self.create_item("ITEM"))
         self.multiworld.push_precollected(self.create_item("FIGHT"))
         self.multiworld.push_precollected(self.create_item("ACT"))
         self.multiworld.push_precollected(self.create_item("MERCY"))
-        if self.multiworld.route_required[self.player] == "genocide":
+        if self.options.route_required == "genocide":
             itempool = [item for item in itempool if item != "Popato Chisps" and item != "Stained Apron" and
                         item != "Nice Cream" and item != "Hot Cat" and item != "Hot Dog...?" and item != "Punch Card"]
-        elif self.multiworld.route_required[self.player] == "neutral":
+        elif self.options.route_required == "neutral":
             itempool = [item for item in itempool if item != "Popato Chisps" and item != "Hot Cat" and
                         item != "Hot Dog...?"]
-        if self.multiworld.route_required[self.player] == "pacifist" or \
-                self.multiworld.route_required[self.player] == "all_routes":
+        if self.options.route_required == "pacifist" or \
+                self.options.route_required == "all_routes":
             itempool += ["Undyne Letter EX"]
         else:
             itempool.remove("Complete Skeleton")
             itempool.remove("Fish")
             itempool.remove("DT Extractor")
             itempool.remove("Hush Puppy")
-        if self.multiworld.key_hunt[self.player]:
-            itempool += ["Key Piece"] * self.multiworld.key_pieces[self.player].value
+        if self.options.key_hunt:
+            itempool += ["Key Piece"] * self.options.key_pieces.value
         else:
             itempool += ["Left Home Key"]
             itempool += ["Right Home Key"]
-        if not self.multiworld.rando_love[self.player] or \
-                (self.multiworld.route_required[self.player] != "genocide" and
-                 self.multiworld.route_required[self.player] != "all_routes"):
+        if not self.options.rando_love or \
+                (self.options.route_required != "genocide" and
+                 self.options.route_required != "all_routes"):
             itempool = [item for item in itempool if not item == "LOVE"]
-        if not self.multiworld.rando_stats[self.player] or \
-                (self.multiworld.route_required[self.player] != "genocide" and
-                 self.multiworld.route_required[self.player] != "all_routes"):
+        if not self.options.rando_stats or \
+                (self.options.route_required != "genocide" and
+                 self.options.route_required != "all_routes"):
```

Comment:
Same thing here, wonder if these could be reformatted to look nicer now

Reply:
These lines could only have the line separation removed for the route_required parentheses section